src/main/java/com/yahoo/bullet/querying/QueryManager.java

```diff
@@ -0,0 +1,252 @@
+/*
+ *  Copyright 2018, Yahoo Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.querying;
+
+import com.yahoo.bullet.common.BulletConfig;
+import com.yahoo.bullet.parsing.Query;
+import com.yahoo.bullet.querying.partitioning.Partitioner;
+import com.yahoo.bullet.record.BulletRecord;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.math3.stat.descriptive.moment.StandardDeviation;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * This class wraps the concept of the {@link Partitioner} and allows you to control using your configured (or not)
+ * partitioner, what queries are seen for each {@link BulletRecord}. It also uses the {@link QueryCategorizer} to
+ * categorize your queries for you if you use the {@link #categorize()} (for categorizing all queries) or
+ * {@link #categorize(BulletRecord)} (for categorizing all partitioned queries for a record). You can get the queries
+ * relevant to your record (after applying any partitioner) using {@link #partition(BulletRecord)}. You can use the
+ * {@link #addQuery(String, Querier)} to add a query to the manager and the {@link #removeAndGetQuery(String)} and
+ * {@link #removeQueries(Set)} methods to remove a query from the manager.
+ */
+@Slf4j
+public class QueryManager {
+    private Map<String, Set<String>> partitioning;
+    private Map<String, Querier> queries;
+    private Partitioner partitioner;
+    private long queriesSeen = 0;
```

Comment:
how will you reset these numbers? it will eventually overflow.

Reply:
I did the math: if you have 1 million rps and 10000 queries, you will increment this by 10 billion each sec. So (2^63 - 1)/(10^10 * 3600 * 24 * 365) is 29 years. I think we're good.

Reply:
👍